### PR TITLE
wharf-core logging & configs

### PR DESCRIPTION
**Pull request by applejag**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Swaggo/swag v1.7.1, replacing dependency on alecthomas/template with stdlib text/template.
- Added logging via wharf-core/pkg/logger, instead of via `fmt.Println`
- Added loading configuration via wharf-core/pkg/config, thus allowing both config file and environment variables

## Motivation

Keeping the projects up to date with our own packages.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	// (v3.0.0), the environment variable BIND_ADDRESS, which was added in
+	// v2.0.0, will also set this value.
+	//
+	// Added in v1.3.0.
+	BindAddress string
+}
+
+// CORSConfig holds settings for the HTTP server's CORS settings.
+type CORSConfig struct {
+	// AllowAllOrigins enables all
```
Comment (Alexamakans) [inline config.go]: It seems like part of the comment has gotten chopped here.

Comment (applejag) [reply config.go]: Sure does. It seems to be this way in all regarded repos.

What about this:

```suggestion
	// AllowAllOrigins enables CORS and allows all hostnames and URLs in the
	// HTTP request origins when set to true. Practically speaking, this
	// results in the HTTP header "Access-Control-Allow-Origin" set to "*".
```

? I will apply it in all api repos

Comment (Alexamakans) [reply config.go]: Looks good 👍

**Review by Alexamakans — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -30,64 +34,58 @@ const ProviderName = "gitlab"
...
-func newWharfClient(authHeader string) wharfapi.Client {
-	return wharfapi.Client{
-		ApiUrl:     os.Getenv("WHARF_API_URL"),
-		AuthHeader: authHeader,
+	if err := r.Run(config.HTTP.BindAddress); err != nil {
+		log.Error().
+			WithError(err).
+			WithString("address", config.HTTP.BindAddress).
+			Message("Failed to start web server.")
+		os.Exit(2)
```
Comment (fredx30) [inline main.go]: Magic number here. Refactor to named constant.

**Review by fredx30 — APPROVED** [review]
This is doing way more than the description lets on which is dangerous. If this was only a package update i would expect only dockerfile, gosum and gomod to change.
Code quality seems good. have not checked for intent as per agreement for the logging project.

**Comment by applejag** [comment]
> @fredx30  This is doing way more than the description lets on which is dangerous. If this was only a package update i would expect only dockerfile, gosum and gomod to change.
> Code quality seems good. have not checked for intent as per agreement for the logging project.

Oh I thought it was enough with the

> - wharf-core logging.
> - wharf-core configs.

I'll update it so it more describes what's up

**Comment by applejag** [comment]
> @fredx30 have not checked for intent as per agreement for the logging project.

That verbal agreement was on the problem/error responses. What kind of declaration of intent would you like to see?

**Pull request merged**